code/obj/item/plants.dm

```diff
@@ -484,6 +487,19 @@
 		..()
 		return
 
+/obj/item/plant/flower/rose/holorose
+	name = "holo rose"
+	desc = "A holographic display of a Rose. This one likes to be called "
+	icon_state = "holorose"
+	backup_name_txt = "names/ai.txt"
+
+	possible_rose_names()
+		var/list/possible_names = list()
+		for(var/mob/M in mobs)
```

Comment:
this can be filtered down to silicon

Reply:
done!

Reply:
Either do

```
for(var/mob/M in mobs)
    if(isAI(M) || isrobot(M))
        possible_names += M
``` 
or 
```
for(var/mob/living/silicon/M in mobs)
    possible_names += M
```
Doing both technically involves an extra check, since under the hood it's checking for `/mob/living/silicon/M` first.

Reply:
Yeah I originally wanted to keep the first method to avoid capturing ghostdrones so I wasn't sure what to do with changing it to /living/silicons and still keep that check for just AI and borgs